File: 01-Threads-and-Locks/john/ex2-Reorder/Reorder.java

public class Reorder {
  	
  	public static void main(String[] args) throws InterruptedException {
		
		while(! Printer.done){
			RunOnce();				
		}
	}

	static int count = 35;
	static void RunOnce() throws InterruptedException {
  		Printer.runCount++;

  		Printer printer = new Printer();		

		Thread buildAndPrint = new Thread(){
			public void run()  {
				printer.BuildAndPrint(count);
			};
		};

		Thread watch = new Thread(){
			public void run() {
				printer.Watch(count);
			};
		};

		watch.start();
		buildAndPrint.start();
		buildAndPrint.join();
		watch.join();
  	}
}

File: 01-Threads-and-Locks/alessandro/Day 1 - Reorder/Reorder.cs
using System;
using System.Threading;

namespace Concurrency
{
	class Reorder
	{
		public static void Main (string[] args)
		{
			int x = 0;
			int y = 0;

			var t1 = new Thread (() => 
			{
				while (true) 
				{
					x++;
					y++;
				}
			});

			var t2 = new Thread (() => 
			{
				while (true) 
				{
					if (x < y)
					{
						Console.WriteLine("Oh, dear!");
					}
				}
			});

			t1.Start ();
			t2.Start ();

			t1.Join ();
			t2.Join ();
		}
	}
}
